dashboard/src/pages/ingest/upload/index.vue

```diff
@@ -0,0 +1,72 @@
+<script setup lang="ts">
+import Uppy from "@uppy/core";
+import { Dashboard } from "@uppy/vue";
+import XHR from "@uppy/xhr-upload";
+import { useRouter } from "vue-router/auto";
+
+import { getPath } from "@/client";
+import { useAuthStore } from "@/stores/auth";
+import { useLayoutStore } from "@/stores/layout";
+import IconUpload from "~icons/clarity/plus-circle-line";
+
+import "@uppy/core/dist/style.css";
+import "@uppy/dashboard/dist/style.css";
+import "@uppy/progress-bar/dist/style.css";
+
+const router = useRouter();
+const authStore = useAuthStore();
+const layoutStore = useLayoutStore();
+
+const GiB = 1024 ** 3; // 1 GiB in bytes
+const maxFileSize = 500 * GiB;
+
+layoutStore.updateBreadcrumb([{ text: "Ingest" }, { text: "Upload SIPs" }]);
+
+const uppy = new Uppy({ restrictions: { maxFileSize: maxFileSize } }).use(XHR, {
+  endpoint: getPath() + "/ingest/sips/upload",
+  allowedMetaFields: false,
+  // Called again for every retry too.
+  async onBeforeRequest(xhr) {
+    if (!authStore.getUserAccessToken) {
+      await authStore.signinSilent();
+    }
+    xhr.setRequestHeader(
+      "Authorization",
+      `Bearer ${authStore.getUserAccessToken}`,
+    );
+  },
+  async onAfterResponse(xhr) {
+    switch (xhr.status) {
+      // "202 Accepted" is returned on successful upload.
+      case 202:
+        setTimeout(() => {
+          router.push({
+            path: "/ingest/sips",
+          });
+        }, 500);
+        break;
+      // "401 Unauthorized" is returned when the auth token has expired.
+      case 401:
+        await authStore.signinSilent();
+        break;
```

Comment:
If we do it before request, why do we need it after response?

Reply:
But doing it after response won't fix that issue. If `onBeforeRequest` is called on retries too (as they said), it doesn't make sense to me to do it in `onAfterResponse`.